### PR TITLE
ActionMenu upstream from Experimental

### PR DESCRIPTION
**Pull request by langermank**
### Description

- Upstream from Experimental
- Add single and multi select functionality

Closes https://github.com/github/primer/issues/1828
Closes https://github.com/github/primer/issues/797

**Remaining TODO**

- [x] If item contains `href`, change `li` role to `role="none"` and `a` role to `role="menuitem"`
- [x] General cleanup
- [x] Tests
- [x] Bonus: use ActionList component (if this is doable, we need to add the select functionality to ActionList instead)

### Integration

> Does this change require any updates to code in production?

No

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews

### CODE REVIEW

**Comment by langermank** [comment]
Related issue here we should try and address https://github.com/github/primer/issues/1898

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "@github/clipboard-copy-element": "^1.1.2",
     "@github/details-menu-element": "^1.0.12",
     "@github/image-crop-element": "^5.0.0",
+    "@github/include-fragment-element": "^6.1.1",
```
Comment (jonrohan) [inline package.json:53]: Is this used anywhere?

Comment (camertron) [reply package.json:53]: @jonrohan I just added an import for it, but it's used to defer load of menu items.

File: app/components/primer/alpha/action_list/item.rb [hunk 2]
```diff
@@ -235,8 +238,7 @@ def initialize(
         def before_render
           @system_arguments[:classes] = class_names(
             @system_arguments[:classes],
-            "ActionListItem--withActions" => trailing_action.present?,
-            "ActionListItem--navActive" => active?
+            "ActionListItem--withActions" => trailing_action.present?
```
Comment (langermank) [inline app/components/primer/alpha/action_list/item.rb:241]: How does this change impact NavList?

Comment (camertron) [reply app/components/primer/alpha/action_list/item.rb:241]: `NavList` adds this class so it's not necessary for `ActionList` to do it.

File: app/components/primer/alpha/action_menu/action_menu_element.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  set dynamicLabel(value: boolean) {
+    this.toggleAttribute('data-dynamic-label', value)
+  }
+
+  get popoverElement(): HTMLElement | null {
+    return this.querySelector<HTMLElement>('[popover]')
+  }
+
+  get invokerElement(): HTMLElement | null {
+    const id = this.querySelector('[role=menu]')?.id
```
Comment (langermank) [inline app/components/primer/alpha/action_menu/action_menu_element.ts:55]: Right now, we have two `role="menu"` declarations. One on the custom element, and another on the `ul`. I _think_ it should only be on the `ul` as its the direct parent of the items, but I could be wrong. It just shouldn't be in both places (that I know for sure)

File: app/components/primer/alpha/action_menu/action_menu_element.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+          }
+        }
+        this.#setDynamicLabel()
+      }
+      event.preventDefault()
+      this.popoverElement?.hidePopover()
+    }
+  }
+
+  #setDynamicLabel() {
```
Comment (langermank) [inline app/components/primer/alpha/action_menu/action_menu_element.ts:107]: @keithamus I'm not sure what happened, but I couldn't get this working 🤔 I added the style for the label and fixed the button gap without visually testing it

File: app/components/primer/alpha/action_menu/action_menu_element.ts (hunk 3 above)
Comment (keithamus) [reply app/components/primer/alpha/action_menu/action_menu_element.ts:55]: `Overlay` (`<anchored-position popover>`) requires either `role=dialog` or `role=menu`. `ActionMenu::List` (`<ul>`) enforces `role=menu`. We can make Overlay's role use optional or omit the enforcement in `ActionMenu::List`.

File: app/components/primer/alpha/action_list/item.html.erb [hunk 5]
```diff
@@ -5,6 +5,15 @@
         <%= private_leading_action_icon %>
       </span>
     <% end %>
+    <% if list.select_variant == :single || list.select_variant == :multiple %>
+      <span class="ActionListItem-visual ActionListItem-action--leading">
+        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16" class="ActionListItem-singleSelectCheckmark"><path fill-rule="evenodd" d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"></path></svg>
```
Comment (jonrohan) [inline app/components/primer/alpha/action_list/item.html.erb:10]: Curious why these aren't `Primer::Beta::Octicon`?

File: app/components/primer/alpha/action_menu.rb [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+        @system_arguments[:"data-dynamic-label-prefix"] = dynamic_label_prefix if dynamic_label_prefix.present?
+
+        @overlay = Primer::Alpha::Overlay.new(
+          id: @menu_id,
+          title: "Menu",
+          visually_hide_title: true
+        )
+
+        @list = Primer::Alpha::ActionMenu::List.new(
+          menu_id: menu_id,
```
Comment (jonrohan) [inline app/components/primer/alpha/action_menu.rb:281]:
```suggestion
          menu_id: @menu_id,
```

File: app/components/primer/beta/button.pcss [hunk 7]
```diff
@@ -27,7 +27,7 @@
   flex-direction: row;
   justify-content: space-between;
   align-items: center;
-  gap: var(--primer-control-medium-gap, 8px);
+  gap: var(--base-size-4, 4px);
```
Comment (jonrohan) [inline app/components/primer/beta/button.pcss:30]: Is this a significant change for all our buttons?

Comment (langermank) [reply app/components/primer/beta/button.pcss:30]: It actually hasn't been doing anything at all 🙈 I think it was left in by mistake. All buttons have the `content` layer inside that controls layout/gap. But we found a use case for it here where we render a label outside of that container I think? I'll double check that.

**Review by camertron — APPROVED** [review]

**Pull request merged**